echion/core.pyi

```diff
@@ -21,6 +22,13 @@ def init_asyncio(
 ) -> None: ...
 def link_tasks(parent: Task, child: Task) -> None: ...
 
+# Greenlet support
+def track_greenlet(
+    greenlet_id: int, name: str | None, frame_cell: list[FrameType | None | bool]
```

Comment:
I'm not that familiar with how `greenlet` works in general, but from what I read, `frame_cell` is simply a list of frames for this specific `greenlet`.
1. it's empty when it hasn't yet started and will be empty again when if finishes executing
2. it will have a list of frames while executing
according to the doc https://greenlet.readthedocs.io/en/stable/greenlet.html

but looking at the implementation in greenlets.h and gevent.py, guess we need to set a sentinel of type bool to properly unwind the stack.

Reply:
The frame cell is a list that will always have one element. This can be either an actual frame (when a greenlet that is still active yields control to another greenlet), or `None` (the target greenlet will always have `None`, and then we will have `None` for a finished origin greenlet). In the native setting, we use `None` as a signal that we need to grab the frame from the thread state. Therefore we need a sentinel object for when a frame is actually not set (because the greenlet has not started/has finished). The simplest thing that I thought of was to just use an existing "sentinel" object, i.e. `False`.